test/widgets/new_dm_sheet_test.dart

```diff
@@ -0,0 +1,146 @@
+
+      check(find.text('Back')).findsOne();
+      check(find.text('Next')).findsOne();
+      check(find.byIcon(ZulipIcons.chevron_left)).findsOne();
+      check(find.byIcon(ZulipIcons.chevron_right)).findsOne();
+
+      final nextButton = tester.widget<GestureDetector>(
+        find.widgetWithText(GestureDetector, 'Next'));
+      check(nextButton.onTap).isNull();
+    });
+
+    testWidgets('shows filtered users based on search', (tester) async {
+      final users = [
+        eg.user(userId: 1, fullName: 'Alice Anderson'),
+        eg.user(userId: 2, fullName: 'Bob Brown'),
+        eg.user(userId: 3, fullName: 'Alice Carter'),
+      ];
+      await setupSheet(tester, users: users);
+
+      await tester.enterText(find.byType(TextField), 'Alice');
+      await tester.pump();
+      check(find.text('Alice Anderson')).findsOne();
+      check(find.text('Alice Carter')).findsOne();
+      check(find.text('Bob Brown')).findsNothing();
+    });
+
+    testWidgets('can select and unselect users', (tester) async {
+      final user = eg.user(userId: 1, fullName: 'Test User');
+      await setupSheet(tester, users: [user]);
+
+      check(find.byIcon(Icons.circle_outlined)).findsOne();
+      check(find.byIcon(Icons.check_circle)).findsNothing();
+
+      final userTile = find.ancestor(
+        of: find.text(user.fullName),
+        matching: find.byType(InkWell),
+      );
+
+      await tester.tap(userTile);
+      await tester.pump();
+      check(find.byIcon(Icons.check_circle)).findsOne();
+      check(find.byIcon(Icons.circle_outlined)).findsNothing();
+
+      final nextButton = tester.widget<GestureDetector>(
+        find.widgetWithText(GestureDetector, 'Next'));
+      check(nextButton.onTap).isNotNull();
+
+      await tester.tap(userTile);
+      await tester.pump();
+      check(find.byIcon(Icons.circle_outlined)).findsOne();
+      check(find.byIcon(Icons.check_circle)).findsNothing();
+    });
+
+    Future<void> runAndCheck(WidgetTester tester, {
+      required User user,
+      NavigatorObserver? navigatorObserver,
+    }) async {
+      final pushedRoutes = <Route<dynamic>>[];
+      final testNavObserver = TestNavigatorObserver()
+        ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+
+      await setupSheet(tester,
+        users: [user],
+        navigatorObserver: testNavObserver);
+
+      final userTile = find.ancestor(
+        of: find.text(user.fullName),
+        matching: find.byType(InkWell),
+      );
+
+      await tester.tap(userTile);
+      await tester.pump();
+      await tester.tap(find.widgetWithText(GestureDetector, 'Next'));
+
+      check(pushedRoutes).last.isA<WidgetRoute>().page
+        .isA<MessageListPage>()
+        .initNarrow.equals(DmNarrow.withOtherUsers(
+          [user.userId],
+          selfUserId: eg.selfUser.userId));
+    }
+
+    testWidgets('navigates to message list on Next', (tester) async {
+      final user = eg.user(userId: 1, fullName: 'Test User');
```

Comment:
We can skip specifying `userId` because we don't rely on it being this particular value for the test.